components/Competition/CreateCompetition/index.js

```diff
@@ -26,42 +36,58 @@ export default function CreateCompetitionForm() {
     setCompetitionGameInput(Number(event.target.value));
   }
 
-  function handleSubmit(event) {
-    event.preventDefault();
+  function handleSubmit() {
     handleAddCompetition(competitionNameInput, competitionGameInput);
 
     router.push("/");
   }
 
   return (
     <>
-      <StyledDetailsLink href={"/"}>Cancel</StyledDetailsLink>
-      <StyledForm onSubmit={handleSubmit}>
-        <StyledFormLabel htmlFor="competition-name">Name</StyledFormLabel>
-        <StyledFormInput
+      <StyledLinkComponent type="back" href={"/"}>
+        Cancel
+      </StyledLinkComponent>
+      <PageHeadlineComponent>Create Competition</PageHeadlineComponent>
+      <StyledPageDescription>
+        Give the new competition a name and define the number of games. After
+        submit it will appear on <strong>Home</strong>.
+      </StyledPageDescription>
+      <StyledForm onSubmit={(event) => event.preventDefault()}>
+        <label htmlFor="competition-name">Name</label>
+        <input
           type="text"
           name="competition-name"
           id="competition-name"
           pattern="^(?!\s*$).+"
+          maxLength={50}
           value={competitionNameInput}
           onChange={handleNameInput}
           required
         />
         <StyledFormLabelInputWrapper>
-          <StyledFormLabel htmlFor="competition-games">
-            Number of Games
-          </StyledFormLabel>
-          <StyledFormInput
+          <label htmlFor="competition-games">Number of Games</label>
+          <input
             type="number"
             name="competition-games"
             id="competition-games"
             value={competitionGameInput}
             onChange={handleGameInput}
             min={1}
             max={100}
+            required
           />
         </StyledFormLabelInputWrapper>
-        <StyledFormButton>Create competition</StyledFormButton>
+        <StyledButtonComponent
+          type="add"
+          callback={handleSubmit}
```

Comment:
`callback` is not a good prop name - it's not clear which event triggers this. Is it simply `onClick` or something different?